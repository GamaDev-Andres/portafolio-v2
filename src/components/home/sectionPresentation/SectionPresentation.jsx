import FigureFoto from './FigureFoto'
import ListOfLinksContact from './ListOfLinksContact'
const SectionPresentation = () => {
  return (
    <section className="md:flex md:justify-center md:items-center md:gap-8 min-h-screen block ">
      <FigureFoto />
      <ListOfLinksContact />
    </section>
  )
}

export default SectionPresentation
