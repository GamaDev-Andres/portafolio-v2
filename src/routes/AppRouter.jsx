import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
