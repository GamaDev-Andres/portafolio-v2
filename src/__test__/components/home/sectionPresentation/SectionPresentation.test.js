/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import SectionPresentation from '../../../../components/home/sectionPresentation/SectionPresentation'
import AppRouter from '../../../../routes/AppRouter'

describe('probando <SectionPresentation/>', () => {
  const wrapper = Enzyme.mount(
    <SectionPresentation />
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('FigureFoto').length).toBe(1)
    expect(wrapper.find('ResumenDescription').exists()).toBe(true)
  })
  test('No debe montarse en ruta "project"', () => {
    const wrapper = Enzyme.mount(
      <HelmetProvider>
        <MemoryRouter initialEntries={['/project/BARBERAPP']}>
          <AppRouter />
        </MemoryRouter>
      </HelmetProvider>
    )
    expect(wrapper.find('SectionPresentation').exists()).toBe(false)
  })
})
