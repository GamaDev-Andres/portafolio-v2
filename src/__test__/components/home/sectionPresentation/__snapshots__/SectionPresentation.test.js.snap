// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <SectionPresentation/> debe montarse correctamente 1`] = `
<section
  className="sm:flex sm:justify-center sm:items-center md:gap-8 sm:min-h-screen block sm:scroll-m-24 md:scroll-m-14"
  id="inicio"
>
  <figure
    className="aspect-square p-4 max-w-xs mx-auto rounded-full"
  >
    <img
      alt="foto Andres Gama"
      className="w-full object-cover object-top  shadow-2xl shadow-blue-200 h-full rounded-full"
      src={Object {}}
    />
  </figure>
  <div
    className="max-w-2xl"
  >
    Array [
      <span
        className="uppercase font-medium text-xl text-fondos"
      >
        Hola soy
      </span>,
      <h2
        className="my-4 uppercase font-bold text-3xl "
      >
        <span
          className="text-blue-400"
        >
          andres
        </span>
         gama
      </h2>,
      <p
        className="whitespace-pre-wrap text-fondos"
      >
        Un
        <span
          className="text-blue-400 font-medium"
        >
           Fullstack Web Developer 
        </span>
        apasionado por el mundo del desarrollo y por crear aplicaciones e interfaces que aporten y brinden grandes experiencias.
      </p>,
    ]
    <div
      className="py-4 gap-4 flex justify-start items-center flex-wrap"
    >
      <a
        className=" text-sm border border-solid shadow-blue-200 shadow-lg border-blue-400 px-6 py-2 rounded-full"
        download="Andres-Gama-CV-Fullstack-developer.pdf"
        href={Object {}}
      >
        Resum√©
      </a>
      <a
        className="p-2 flex flex-col justify-center items-center shadow-md shadow-blue-200 border border-solid border-blue-100"
        href="https://github.com/GamaDev-Andres"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fa-brands fa-github"
        />
        <span
          className="text-xs"
        >
          Github
        </span>
      </a>
      <a
        className="p-2 flex flex-col justify-center items-center shadow-md shadow-blue-200 border border-solid border-blue-100"
        href="https://www.linkedin.com/in/andres-gama/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="fa-brands fa-linkedin"
        />
        <span
          className="text-xs"
        >
          LinkedIn
        </span>
      </a>
    </div>
  </div>
</section>
`;
